concepts/for-loops/about.md

```diff
@@ -1,61 +1,67 @@
-# About
+# About For Loops
 
-The [for loop][for-loop] provides a mechanism to execute a group of statements repeatedly.
+The [for loop](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html) provides a mechanism to execute a group of statements repeatedly until some condition is met.
 The loop consists of four parts:
 
 ```java
-for (initialization; test; update)
-{
-     // body
+for (initialization; test; update) {
+    body;
 }
 ```
 
-The initialization sets an initial state for the loop.
-Typically it declares and sets a variable used in the test expression and update statement.
+The `initialization` sets an initial state for the loop and is executed exactly once at the start of the loop.
+Typically it declares and assigns a variable used in the test expression and update statement.
 For example:
 
 ```java
-int i=1
+int i = 1
 ```
 
-The test expression tests if the loop should execute the body
-and execute the update statement.
-
-If the test evaluates to true the body and the update expression will be executed.
-
-If the expression evaluates to false neither the body nor the update statement will be executed and execution continues after the loop.
-An example of a test can be:
+The `test` expression tests if the loop should end.
+If it evaluates to true, the body and then the update expression will be executed.
+If it evaluates to false, neither the body nor the update statement will be executed and execution resumes after the loop's closing bracket.
+Typically it checks the variable assigned in the initialization block.
+For example:
 
 ```java
 i <= 10
 ```
 
-After executing the loop body, the update expression increments/decrements the loop variable by some value.
-Example:
+After executing the loop body, the `update` expression is executed.
+Typically it increments or decrements the loop variable by some value.
+For example:
 
 ```java
 i++
 ```
 
-A for loop executing over each element in an array can look like this:
+A for loop printing out the first four squares would look like this:
 
 ```java
-char[] vowels = {'a', 'e', 'i', 'o', 'u'};
-
-for (int i = 0; i<vowels.length; i++) {
-    // Output the vowel
-    System.out.print(vowels[i]);
+for (int i = 1; i <= 4; i++) {
+    System.out.println(i * i);
 }
 
-// => aeiou
+/* =>
+1
+4
+9
+16
+*/
 ```
 
-A `for` loop does have some advantages over a `foreach` loop:
+If iterating through every element in a collection, a for-each loop is preferred, but it can be done with a for loop like this:
```

Comment:
```suggestion
If iterating through every element in a collection, a `for-each` loop is preferred, but it can be done with a `for` loop like this:
```